clowder-framework/clowder2
# Geospatial Preview

To test this PR:

1) build the previewer container
- Switch to this branch of the `extractors-geo` repo: https://github.com/clowder-framework/extractors-geo/pull/29/files
- In preview.geotiff folder, run `docker build -f Dockerfile -t clowder/extractors-geotiff-preview .`

2) run the geoserver + previewer docker compose
- In clowder2 main directory, run `docker compose -f docker-compose.previewers.yml up -d`

3) upload a geotiff file
- I found one here: https://download.osgeo.org/geotiff/samples/
- The previewer should upload the file to geoserver, then add file metadata to the geotiff.
- Refreshing file view in Clowder should show a zoom to the image, with OSM map behind:
![image](https://github.com/clowder-framework/clowder2/assets/14166971/0c47c088-2e58-442a-b85d-4f32a9d5b5d5)




UPDATE:
- Need to modify this to look at vis config instead.
- https://github.com/clowder-framework/pyclowder/pull/71 need this to be merged, and update the extractor to write to the config for v2: https://github.com/clowder-framework/extractors-geo/pull/29
